Src/HALALMock/Services/Encoder/Encoder.cpp

```diff
@@ -22,128 +22,73 @@ uint8_t Encoder::inscribe(Pin& pin1, Pin& pin2) {
             pin1.to_string().c_str(), pin2.to_string().c_str());
         return 0;
     }
-
-    Pin::inscribe(pin1, ALTERNATIVE);
-    Pin::inscribe(pin2, ALTERNATIVE);
-
     uint8_t id = Encoder::id_counter++;
     Encoder::registered_encoder[id] = doublepin;
 
+    EmulatedPin &pin1_data = SharedMemory::get_pin(pin1);
+    EmulatedPin &pin2_data = SharedMemory::get_pin(pin2);
+
+	if(pin1_data.type == PinType::NOT_USED && pin2_data.type == PinType::NOT_USED){
+		pin1_data.type = PinType::ENCODER;
+        pin2_data.type = PinType::ENCODER;
+        pin1_data.PinData.ENCODER.direction = false;
+		pin1_data.PinData.ENCODER.count_value = 0;
+        pin1_data.PinData.ENCODER.is_on = false;
+    }else{
+        ErrorHandler("Pin1:%s or Pin2:%s are being used already",pin1.to_string(),pin2.to_string());
+    }
     return id;
 }
 
-void Encoder::start() {
-    for (pair<uint8_t, pair<Pin, Pin>> instance : registered_encoder) {
-        init(Encoder::pin_timer_map[instance.second]);
-    }
-}
+void Encoder::start(){}
 
 void Encoder::turn_on(uint8_t id) {
     if (not Encoder::registered_encoder.contains(id)) {
         ErrorHandler("No encoder registered with id %u", id);
         return;
     }
-
-    TimerPeripheral* timer = pin_timer_map[registered_encoder[id]];
-
-    if (HAL_TIM_Encoder_GetState(timer->handle) == HAL_TIM_STATE_RESET) {
-        ErrorHandler("Unable to get state from encoder in timer %s",
-                     timer->name.c_str());
-        return;
-    }
-
-    if (HAL_TIM_Encoder_Start(timer->handle, TIM_CHANNEL_ALL) != HAL_OK) {
-        ErrorHandler("Unable to start encoder in timer %s",
-                     timer->name.c_str());
-        return;
-    }
-
-    reset(id);
+    std::pair<Pin,Pin> pair_pin = Encoder::registered_encoder[id];
+    EmulatedPin &pin1_data = SharedMemory::get_pin(pair_pin.first);
+    pin1_data.PinData.ENCODER.is_on = true;
+    
 }
 
 void Encoder::turn_off(uint8_t id) {
     if (not Encoder::registered_encoder.contains(id)) {
         ErrorHandler("No encoder registered with id %u", id);
         return;
     }
-
-    TimerPeripheral* timer = pin_timer_map[registered_encoder[id]];
-
-    if (HAL_TIM_Encoder_Stop(timer->handle, TIM_CHANNEL_ALL) != HAL_OK) {
-        ErrorHandler("Unable to stop encoder in timer %s", timer->name.c_str());
-    }
+    std::pair<Pin,Pin> pair_pin = Encoder::registered_encoder[id];
+    EmulatedPin &pin1_data = SharedMemory::get_pin(pair_pin.first);
+    pin1_data.PinData.ENCODER.is_on = false;
 }
 
-void Encoder::reset(uint8_t id) {
+void Encoder::reset(uint8_t id) {}
```

Comment:
set the value to the one the original code is saying, half of the representable range (e.g. for uint32_t -> UINT32_MAX / 2)